ScarletEngine/Sources/Gameplay/Public/World.h

```diff
@@ -45,6 +45,11 @@ namespace ScarletEngine
 			return *static_cast<System<SystemSig...>*>(Systems.emplace_back(GlobalAllocator<System<SystemSig...>>::New(&Reg, Name)).get());
 		}
 
+		void AddSystem(ISystem* System)
+		{
+			Systems.emplace_back(System);
```

Comment:
We should do a check here to ensure that the passed system is non-null.

You could either do it with an if statement and silently ignore it or probably better would be to use a `check (System != nullptr)` to crash the engine if this ever happens.